alerter.c

```diff
@@ -23,10 +28,47 @@ void alertInCelcius(float farenheit) {
     }
 }
 
-int main() {
+float ConvertToCelcius(float farenheit) 
+{
+    float celcius = (farenheit - 32) * 5 / 9;
+    printf("\nconverted: %.1f celcius.\n",celcius);
+    return celcius;
+}
+
+int test(float Temperature)
+{
+    float celcius = ConvertToCelcius(Temperature);
+    int FAIL = networkAlertStub(celcius);
+    printf("Test case :\n");
+    
+    if(FAIL == 200)
+    {
+        //printf("OK: Good condition: Temperature is %.1f celcius.\n", FAIL);
+        printf("OK: ALERT: Temperature is %d celcius.\n", FAIL);
+         NotFailureCount += 1;
+        printf("not FailureCount is %d\n", NotFailureCount);
+    }
+    else if(FAIL == 500)
+    {
+        //printf("NOT OK: ALERT: Temperature is %.1f celcius.\n", FAIL);
+        printf("NOT OK: ALERT: Temperature is %d celcius.\n", FAIL);
+        FailureCount += 1;
+        printf("FailureCount is %d\n", FailureCount);
+    }
+    return FAIL;
+}
+
+int main() 
+{
     alertInCelcius(400.5);
     alertInCelcius(303.6);
-    printf("%d alerts failed.\n", alertFailureCount);
+
+    assert(test(300) == 200);
+    assert(test(350) == 200);
+    assert(test(400) == 200);
+    assert(test(800) == 500);
+    printf("\n%d alerts failed.\n", alertFailureCount);

```

Comment:
The software is also supposed to report failure count? in that case an assert to check the failure count too should be included.